rust/agama-lib/src/network/types.rs

```diff
@@ -55,6 +58,71 @@ pub enum DeviceType {
     Bridge = 6,
 }
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum DeviceState {
```

Comment:
> The problem with this `DeviceState` is that it corresponds 100% with the `NetworkManager` one. I would expect something more simple as we are not interesting in all those cases.
> 
> As we do not support more backends, we can leave it as it is by now. But it would be nice to add a comment to make this clear.

I completely agree, by now added them as them are declared and send by NetworkManager, and the same will happen with the state reason and connection events which currently we are not reacting too. We need a deeper analysis of them to handle just the ones we are interested and generalize them too.